improver_tests/acceptance/acceptance.py

```diff
@@ -84,6 +89,125 @@ def cli_name_with_dashes(dunder_file):
+    the sha256sum command line tool.
+
+    Args:
+        checksum_path (Optional[pathlib.Path]): Path to checksum file. File
+            should be plain text in the format produced by the sha256sum
+            command line tool. Paths listed in the file should be relative to
+            the KGO root directory found by kgo_root().
+
+    Returns:
+        Dict[pathlib.Path, str]: Dict with keys being relative paths and
+            values being hexadecimal checksums
+    """
+    if checksum_path is None:
+        checksum_path = DEFAULT_CHECKSUM_FILE
+    if not kgo_exists():
+        raise IOError("KGO path must be defined")
+    with open(checksum_path, mode="r") as checksum_file:
+        checksum_lines = checksum_file.readlines()
+    checksums = {}
+    for line in checksum_lines:
+        parts = line.strip().split("  ", maxsplit=1)
+        csum = parts[0]
+        path = pathlib.Path(parts[1])
+        checksums[path] = csum
+    return checksums
+
+
+def verify_checksum(kgo_path, checksums=None, checksum_path=None):
+    """
+    Verify an individual KGO file's checksum.
+
+    Args:
+        kgo_path (pathlib.Path): Path to file in KGO directory
+        checksums (Optional[Dict[pathlib.Path, str]]): Lookup dictionary
+            mapping from paths to hexadecimal checksums. If provided, used in
+            preference to checksum_path.
+        checksum_path (pathlib.Path): Path to checksum file. File should be
+            plain text in the format produced by the sha256sum command line
+            tool.
+
+    Raises:
+        KeyError: File being verified is not found in checksum dict/file
+        ValueError: Checksum does not match value in checksum dict/file
+    """
+    if checksums is None:
+        checksums = acceptance_checksums(checksum_path)
+    kgo_csum = calculate_checksum(kgo_path)
+    kgo_norm_path = pathlib.Path(os.path.normpath(kgo_path))
+    kgo_rel_path = kgo_norm_path.relative_to(kgo_root())
+    try:
+        expected_csum = checksums[kgo_rel_path]
+    except KeyError:
+        msg = f"Checksum for {kgo_rel_path} is missing"
+        raise KeyError(msg)
+    if kgo_csum != expected_csum:
+        msg = f"Checksum for {kgo_rel_path} is {kgo_csum}, expected {expected_csum}"
+        raise ValueError(msg)
+    return
+
+
+def calculate_checksum(path):
+    """
+    Calculate SHA256 hash/checksum of a file
+
+    Args:
+        path (pathlib.Path): Path to file
+
+    Returns:
+        str: checksum as hexadecimal string
+    """
+    hasher = hashlib.sha256()
+    with open(path, mode="rb") as kgo_file:
+        while True:
+            # read 1 megabyte binary chunks from file and feed them to hasher
+            kgo_chunk = kgo_file.read(2 ** 20)
+            if not kgo_chunk:
+                break
+            hasher.update(kgo_chunk)
+    checksum = hasher.hexdigest()
+    return checksum
+
+
+def verify_checksums(cli_arglist):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Useless return at end of function or method](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5572303)